### ISSUE TITLE
Expo CLI upload doesn't respect owner field of app.json

### ISSUE THREAD

**Issue by chris-soar** [expo/expo-cli#1207]
## Description

Expo CLI `expo upload:ios` doesn't use the owner field of app.json to determine where to download the latest build file from. So if the app is built (`expo build:ios`) with a different owner than the logged in user, the upload command is then unable to find the build on the Expo servers.

## Expected Behavior

The `expo upload:ios` command should retrieve the owner field from app.json (just like it currently retrieves the slug) and use the owner field to search for and download the latest build, if the logged in user has the required privileges. As it stands right now, it's necessary to manually download the build file and add the `--path` option.

## Observed Behavior

The Expo CLI attempted to search for the latest build of the app under the logged in user's builds. As we use delegated team accounts for building the app under the central account's ownership, the CLI wasn't able to find any builds. It seems there isn't a method of directly providing the owner field to the upload command either.
Looking at `_downloadLastestBuild()` in the upload command's `BaseUploader.ts`, it seems the command only retrieves the slug from the config and doesn't retrieve owner. By comparison the iOS build command seems to retrieve the owner and slug from the project's manifest/app.json.

The following error is observed, as Expo CLI doesn't find any builds for the current slug under the logged in user's builds.
```
Failed to upload the standalone app to the app store.
There are no builds on the Expo servers, please run 'expo build:ios' first
```

## Environment

```
Expo CLI 3.4.1 environment info:
    System:
      OS: macOS 10.15.1
      Shell: 5.7.1 - /usr/local/bin/zsh
    Binaries:
      Node: 12.13.0 - ~/.nvm/versions/node/v12.13.0/bin/node
      Yarn: 1.19.1 - /usr/local/bin/yarn
      npm: 6.13.0 - ~/.nvm/versions/node/v12.13.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.5900203
      Xcode: 11.2/11B52 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^35.0.0 => 35.0.1 
      react: 16.8.3 => 16.8.3 
      react-native: https://github.com/expo/react-native/archive/sdk-35.0.0.tar.gz => 0.59.8 
      react-navigation: ^4.0.7 => 4.0.10 
    npmGlobalPackages:
      expo-cli: 3.4.1
```

## Reproducible Demo

Within a team, login to Expo CLI as a user (e.g. user1 with Admin permissions) that is not the owner of the team (e.g. owner1) and build an app (`expo build:ios`) using `"owner": "owner1"` field in app.json to build the app under the main account of the team.
Using the same login, try to upload the latest build with `expo upload:ios` without any options. The command fails to find any builds on the Expo servers with the same error as above, since it can't find any builds under user1 for the app with the slug provided.

**Comment by vincentvella** [comment]
Confirmed this behavior today. Repro is accurate.

**Comment by tcdavis** [comment]
Resolved in `expo-cli@3.4.2`
Many thanks @vincentvella for the PR!
